src/native/libs/System.IO.Compression.Native/CMakeLists.txt

```diff
@@ -30,11 +43,8 @@ if (CLR_CMAKE_TARGET_UNIX OR CLR_CMAKE_TARGET_BROWSER OR CLR_CMAKE_TARGET_WASI)
     append_extra_compression_libs(NATIVE_LIBS_EXTRA)
 
     if (CLR_CMAKE_TARGET_BROWSER OR CLR_CMAKE_TARGET_WASI)
-        include(${CLR_SRC_NATIVE_DIR}/external/zlib.cmake)
-        add_definitions(-DINTERNAL_ZLIB)
-        set_source_files_properties(${ZLIB_SOURCES} PROPERTIES COMPILE_DEFINITIONS "${ZLIB_COMPILE_DEFINITIONS}")
-        set_source_files_properties(${ZLIB_SOURCES} PROPERTIES COMPILE_OPTIONS "${ZLIB_COMPILE_OPTIONS}")
-        set(NATIVECOMPRESSION_SOURCES ${ZLIB_SOURCES} ${NATIVECOMPRESSION_SOURCES})
+        set(NATIVECOMPRESSION_SOURCES ${NATIVECOMPRESSION_SOURCES})
+        list(APPEND NATIVE_LIBS_EXTRA zlibstatic)
```

Comment:
@jkoritzinsky I suspect this is the line causing the CI failure "multiple rules generated" from [this comment](https://github.com/dotnet/runtime/pull/102403#issuecomment-2143020722), because NATIVE_LIBS_EXTRA is then consumed in line 77.

Reply:
I think this might actually be due to the same "shared libraries aren't supported on WASM" problem you're getting (about how `-shared` isn't supported). Try solving that first and see if this gets cleared up in the meantime.